feat(provider): add environment variable support for organization config

juwit opened the pull request:
Hi there 👋

Here is another contribution, to make the organization configurable by an environment variable named `CC_ORGANIZATION`.

I also corrected the english spelling of _organization_.

This is separated in two commits :

- fix(provider): correct typo in provider configuration variable
- feat(provider): add environment variable support for organization config

miton18 commented:
Thanks for your work @juwit !
It's OK to add a environment variable for the organization configuration (I was using it in the tests)

But sadly, we will keep ` organisation`  because all the Clever Cloud API use it.
This is an item to talk about when we will refactor the API

juwit commented:
> Thanks for your work @juwit ! It's OK to add a environment variable for the organization configuration (I was using it in the tests)
> 
> But sadly, we will keep ` organisation` because all the Clever Cloud API use it. This is an item to talk about when we will refactor the API

Hi @miton18 
Ok, I saw that there were API call to `/v2/organisation` somewhere in the client code, and thought that using the right name in the provider was not a big deal.
I can revert to `CC_ORGANISATION` if you want, and revert the provider variable name.
Also, I used the `CC_` prefix to make the environment variables consistent with the `CC_OAUTH_*` ones.

juwit commented:
Hi @miton18 
I removed the commit that renamed the provider `organisation` parameter to `organization`.
I also renamed the environement variable to `CC_ORGANISATION` to be consistent with the parameter naming.

miton18 commented:
Hi @juwit I still see some `organization`

juwit commented:
Hi @miton18.
I leaved the provider struct as-is (it already used a field with the right name) , and removed the last `organization` that was in the doc.
